tests/Application/Commands/DeleteStalePaymentDetailsTest.php

```diff
@@ -0,0 +1,178 @@
+        $initDate = new \DateTimeImmutable('now');
+        $testCustomerId = 'cus_aaaaaaaaaaaa11';
+        $testPaymentMethodId = 'pm_aaaaaaaaaaaa13';
+        $testPaymentMethodFingerprint = 'Xt5EWLLDS7FJjR1c';
+
+        $stripeChargesProphecy = $this->prophesize(ChargeService::class);
+        $stripeChargesProphecy->search([
+            'query' => sprintf(
+                'customer:"%s" and payment_method_details.card.fingerprint:"%s" and status:"succeeded"',
+                $testCustomerId,
+                $testPaymentMethodFingerprint,
+            ),
+            'limit' => 100,
+        ])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->buildEmptyCollection());
+
+        $stripePaymentMethodsProphecy = $this->prophesize(PaymentMethodService::class);
+        $stripePaymentMethodsProphecy->all([
+            'customer' => $testCustomerId,
+            'type' => 'card',
+            'limit' => 100,
+        ])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->buildCollectionFromSingleObjectFixture(
+                $this->getStripeHookMock('ApiResponse/pm'),
+            ));
+
+        // One PM should be detached i.e. soft deleted.
+        $stripePaymentMethodsProphecy->detach($testPaymentMethodId)
+            ->shouldBeCalledOnce()
+            ->willReturn($this->getStripeHookMock('ApiResponse/pm'));
+
+        $oneDayBeforeTest = $initDate
+            ->sub(new \DateInterval('P1D'))
+            ->getTimestamp();
+
+        $stripeCustomersProphecy = $this->prophesize(CustomerService::class);
+        $stripeCustomersProphecy->search([
+            'query' => "created<$oneDayBeforeTest and metadata['hasPasswordSince']:null",
+            'limit' => 100,
+        ])
+            ->shouldBeCalledOnce()
+            ->willReturn($this->buildCollectionFromSingleObjectFixture(
+                $this->getStripeHookMock('ApiResponse/customer'),
+            ));
+
+        $stripeClientProphecy = $this->prophesize(StripeClient::class);
+
+        $stripeClientProphecy->charges = $stripeChargesProphecy->reveal();
+        $stripeClientProphecy->customers = $stripeCustomersProphecy->reveal();
+        $stripeClientProphecy->paymentMethods = $stripePaymentMethodsProphecy->reveal();
+
+        $commandTester = new CommandTester($this->getCommand(
+            $stripeClientProphecy,
+            $initDate,
+        ));
+        $commandTester->execute([]);
+
+        $expectedOutputLines = [
+            'matchbot:delete-stale-payment-details starting!',
+            'Deleted 1 payment methods from Stripe, having checked 1 customers',
+            'matchbot:delete-stale-payment-details complete!',
+        ];
+        $this->assertEquals(implode("\n", $expectedOutputLines) . "\n", $commandTester->getDisplay());
+
+        $this->assertEquals(0, $commandTester->getStatusCode());
+    }
+
+    public function testNotDeletingOneUsedCard(): void
+    {
+        $initDate = new \DateTimeImmutable('now');
```

Comment:
Maybe better to use hard coded dates in tests to make it clear that it doesn't matter what time we run the test at? This still passes:

```suggestion
        $initDate = new \DateTimeImmutable('2023-01-01 00:00:00');
```

Same applies to second test.

Reply:
As discussed, work to address this is started for inclusion on another PR shortly